### ISSUE TITLE
Bundle alloy-mixin as release asset

### ISSUE THREAD

**Issue by rfratto** [grafana/alloy#162]
### Request

Bundle `alloy-mixin` as one of the release assets so users can install it without having to clone the repo. 

This should likely be bundled as a rendered set of dashboards in a Zip instead of Jsonnet; not everyone will want to install Jsonnet and/or Grizzly to deploy the dashboards. 

### Use case

Lower the number of steps it takes to gain access to the official Alloy dashboards/alerts.

**Comment by M0NsTeRRR** [comment]
Linked to #184

**Comment by algo7** [comment]
I there a documentation somewhere explaining how to generate the dashboard and recording rules provided under the alloy-mixin directory?

**Comment by Skaronator** [comment]
Currently trying to figure that out myself. It looks like you need jsonnet and [jsonnet-bundler](https://github.com/jsonnet-bundler/jsonnet-bundler).

Then run within the `operations/alloy-mixin` the install command:

```bash
jb install
```

After that, you should be able to generate the dashboards for my understanding but nothing works.

``` bash
$ jsonnet dashboards.libsonnet
{ }
```

``` bash
$ jsonnet grizzly.jsonnet
RUNTIME ERROR: couldn't open import "github.com/grafana/jsonnet-libs/logs-lib/logs/main.libsonnet": no match locally or in the Jsonnet library paths.
	./grizzly/../././dashboards/alloy-logs.libsonnet:2:23-92	thunk <logsDashboard>
	./grizzly/../././dashboards/alloy-logs.libsonnet:12:9-22	thunk <alloyLogs>
	./grizzly/../././dashboards/alloy-logs.libsonnet:36:24-33	object <anonymous>
	./grizzly/dashboards.jsonnet:23:13-42	object <anonymous>
	./grizzly/dashboards.jsonnet:(15:15)-(26:4)	object <anonymous>
	./grizzly/dashboards.jsonnet:(15:15)-(26:4)	object <anonymous>
	During manifestation
```

**Comment by Skaronator** [comment]
Okay, I figured it out using [Grizzly](https://github.com/grafana/grizzly). Not sure if you need the jsonnet bundler from my last post, but this works:

``` bash
grr export -o json grizzly.jsonnet .
```

**Comment by NissesSenap** [comment]
It has been on my TODO to do some upstream PR around this for a while, but sadly I haven't had the time.
After getting some help from a Grafana employee, I managed to get this to work.

First, tell grr to only generate local files
```shell
grr config set only-spec true
# the grafana token and URL won't be used, but they have to be there for grizzly to work
grr config set grafana.url http://localhost:3000/
grr config set grafana.token abcd12345 #this don't matter
```

Time to build alloy specific stuff

```shell
cd operations/alloy-mixin
grr export grizzly.jsonnet output -o json
# grr also supports yaml output, which can be useful if you want the prometheusRules.
grr export grizzly.jsonnet output2 -o yaml
```

Make sure you are using grr [0.4.2 ](https://github.com/grafana/grizzly/releases/tag/v0.4.2)or later, else stuff won't work. See https://github.com/grafana/grizzly/pull/436 for more info.

Hopefully this helps @algo7 and @Skaronator

**Comment by algo7** [comment]
> Okay, I figured it out using [Grizzly](https://github.com/grafana/grizzly). Not sure if you need the jsonnet bundler from my last post, but this works:
> 
> ```shell
> grr export -o json grizzly.jsonnet .
> ```
> 
> Edit: Okay grizzly exports are not grafana dashboards 🙄 but I think the .spec field is the actual dashboard.

Yes. The `spec` part of the manifests generated by `grr` is actually the rules and the dashboards that can be used in Grafana. 

I did 
```bash
cat dashboard_name.json | jq ".spec"
```
and just use the output directly.

**Comment by algo7** [comment]
> It has been on my TODO to do some upstream PR around this for a while, but sadly I haven't had the time. After getting some help from a Grafana employee, I managed to get this to work.
> 
> First, tell grr to only generate local files
> 
> ```shell
> grr config set only-spec true
> # the grafana token and URL won't be used, but they have to be there for grizzly to work
> grr config set grafana.url http://localhost:3000/
> grr config set grafana.token abcd12345 #this don't matter
> ```
> 
> Time to build alloy specific stuff
> 
> ```shell
> cd operations/alloy-mixin
> grr export grizzly.jsonnet output -o json
> # grr also supports yaml output, which can be useful if you want the prometheusRules.
> grr export grizzly.jsonnet output2 -o yaml
> ```
> 
> Make sure you are using grr [0.4.2 ](https://github.com/grafana/grizzly/releases/tag/v0.4.2)or later, else stuff won't work. See [grafana/grizzly#436](https://github.com/grafana/grizzly/pull/436) for more info.
> 
> Hopefully this helps @algo7 and @Skaronator

I managed to figured it out reading grizzly doc. I was spending so much time with `jsonnet` and `jb` when referencing the [Kubernetes Mixin][1]

[1]: https://github.com/kubernetes-monitoring/kubernetes-mixin

Thanks for replying. Should help other people facing the similar issue before the doc comes out.

**Comment by daanschipper** [comment]
The following worked for me with just jsonnet
```shell
jb init
jb install https://github.com/grafana/alloy/operations/alloy-mixin@main
jsonnet -e '(import "vendor/github.com/grafana/alloy/operations/alloy-mixin/mixin.libsonnet").prometheusAlerts' | yq -P 
jsonnet --jpath vendor --jpath vendor/github.com/grafana/jsonnet-libs --jpath vendor/logs-lib -e '(import "vendor/github.com/grafana/alloy/operations/alloy-mixin/mixin.libsonnet").grafanaDashboards'
```

**Comment by iacabezasbaculima** [comment]
> I've yet to get this to work.
> 
> ```
> ❯ grr export -o json grizzly.jsonnet .
> FATAL parse error in 'grizzly.jsonnet': RUNTIME ERROR: couldn't open import "github.com/grafana/grafonnet/gen/grafonnet-v11.0.0/main.libsonnet": no match locally or in the Jsonnet library paths
> 	jsonnet-libs/logs-lib/logs/g.libsonnet:1:1-75	$
> 	jsonnet-libs/logs-lib/logs/dashboards.libsonnet:1:11-33	thunk <g> from <$>
> 	jsonnet-libs/logs-lib/logs/dashboards.libsonnet:12:7-8
> 	dashboards/alloy-logs.libsonnet:(30:22)-(33:67)	+:
> 	dashboards/alloy-logs.libsonnet:36:24-49	object <anonymous>
> 	grizzly/dashboards.jsonnet:23:13-42	object <anonymous>
> 	Field "spec"
> 	Field "alloy-logs.json"
> 	Field "dashboards"
> 	During manifestation
> ```
> 
> I haven't been able to find anything in the Grizzly docs about a library path?
> 
> Edit: I had to run the following before I could get this to work:
> 
> ```
> jb install github.com/grafana/grafonnet/gen/grafonnet-v11.0.0@main
> ```

Indeed this is required - I can confirm below works:

```console
grr config set only-spec true
grr config set grafana.url http://localhost:3000/
grr config set grafana.token abcd12345 #this don't matter

cd operations/alloy-mixin
jb install github.com/grafana/grafonnet/gen/grafonnet-v11.0.0@main
grr export grizzly.jsonnet output -o json
grr export grizzly.jsonnet output2 -o yaml
```

**Comment by thampiotr** [comment]
There's #5074 proposal how we would like to implement this. We also have first PR https://github.com/grafana/alloy/pull/5118 that will include generated JSON dashboards in the source code, so it should already be easier to import.
